opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/websocket/WebsocketAutoConfiguration.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.cloud.websocket;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurationSupport;
+
+import io.opentracing.Tracer;
+import io.opentracing.tag.Tags;
+
+@Configuration
+@ConditionalOnBean(WebSocketMessageBrokerConfigurationSupport.class)
+@ConditionalOnProperty(name = "opentracing.spring.cloud.websocket.enabled", havingValue = "true", matchIfMissing = true)
+public class WebsocketAutoConfiguration {
+
+    @Autowired
+    Tracer tracer;
+
+    @Autowired
+    WebSocketMessageBrokerConfigurationSupport config;
+
+    @Bean
+    public TracingChannelInterceptor tracingInboundChannelInterceptor() {
```

Comment:
Now it looks much better. This also worked https://paste.fedoraproject.org/paste/YcocqvZsm6TcFwlVBCoqzw/ (similar to sleuth...)